src/Components/Server/src/Circuits/CircuitPrerenderer.cs

```diff
@@ -71,8 +76,14 @@ public async Task<ComponentPrerenderResult> PrerenderComponentAsync(ComponentPre
                 Prerendered = true
             });
 
+            var record = JsonSerializer.ToString(new PrerenderedComponentRecord(
+                    circuitHost.CircuitId.Replace("--", ".."), // We need to do this due to the fact that -- is not allowed within HTML comments and HTML doesn't encode '-'.
```

Comment:
Is the expectation that `CircuitId` would never have `..` in them? Or is that having existing '..` also change to dashes isn't problematic? Could we note that expectation as a comment here?

Reply:
We will never have '.' on them because we base64url encode the id and the only non alphanumeric symbols are '-' and '_', dots are safe on the url, so we simply replace '-' with '.'. You could say its our "own custom base64 encoding scheme"

Reply:
👍  Could you make a note of that?